### PR TITLE
fix view parsing problem

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
In this PR, we ignore unknown property in Json when parse table. This PR can also closes https://github.com/pingcap/tispark/issues/942
### What is changed and how it works?
The trick is adding `JsonIgnoreProperties` annotation. 

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

Related changes
 - Need to cherry-pick to the all release branch

### CODE REVIEW

**Comment by birdstorm** [comment]
do we need to add a test?

**Comment by zhexuany** [comment]
@birdstorm I am adding test.

**Comment by zhexuany** [comment]
blocked by https://github.com/pingcap/tidb/issues/11513

**Comment by marsishandsome** [comment]
> blocked by [pingcap/tidb#11513](https://github.com/pingcap/tidb/issues/11513)

@zhexuany why is blocked?

File: core/src/test/scala/org/apache/spark/sql/ViewTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  test("Test View") {
+    tidbStmt.execute(s"drop table if exists $table")
+
+    tidbStmt.execute(s"create table $table(qty INT, price INT);")
+
+    tidbStmt.execute(s"INSERT INTO $table VALUES(3, 50);")
+    tidbStmt.execute(s"CREATE VIEW v AS SELECT qty, price, qty*price AS value FROM $table;")
+
+    judge(s"select * from $table")
```
Comment (marsishandsome) [inline core/src/test/scala/org/apache/spark/sql/ViewTestSuite.scala]: maybe should `reset connection` before select

**Comment by zhexuany** [comment]
@marsishandsome In MySQL, `drop table` for a view is not validate syntax. Once TiDB adopts this behavior, we have to change our code. I am planning ignore unknown Json property first and add test back till TiDB fix the issue.

**Comment by zhexuany** [comment]
```
scala> spark.sql("use test")
res0: org.apache.spark.sql.DataFrame = []

scala> spark.sql("select * from t").show
+---+
| c1|
+---+
|  1|
+---+


scala> spark.sql("desc t").show
+--------+---------+--------+-------+
|col_name|data_type|nullable|comment|
+--------+---------+--------+-------+
|      c1|   bigint|    true|   null|
+--------+---------+--------+-------+


scala> spark.sql("select * from v").show
+---+-----+
| c1|value|
+---+-----+
+---+-----+
```

I tested on my machine, it works fine. The test suite will be added if TiDB fixes the bug

**Comment by birdstorm** [comment]
@zhexuany  I suggest we investigate the issue of Decimal type more carefully.

**Comment by zhexuany** [comment]
@marsishandsome @birdstorm  PTAL

File: core/src/test/scala/org/apache/spark/sql/ViewTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      case _: Exception => cancel
+    }
+
+    tidbStmt.execute(s"create table $table(qty INT, price INT);")
+
+    tidbStmt.execute(s"INSERT INTO $table VALUES(3, 50);")
+    tidbStmt.execute(s"CREATE VIEW v AS SELECT qty, price, qty*price AS value FROM $table;")
+
+    refreshConnections()
+
```
Comment (marsishandsome) [inline core/src/test/scala/org/apache/spark/sql/ViewTestSuite.scala]: better to add an assert `v does not exist in tispark's catalog`

**Comment by marsishandsome** [comment]
description may shoud change `The trick is adding JsonIgnoreProperties annotation.`

File: tikv-client/src/main/java/com/pingcap/tikv/catalog/Catalog.java [hunk 3]
```diff
@@ -85,15 +85,19 @@ public TiDBInfo getDatabase(String name) {
...
     }
 
     public TiTableInfo getTable(TiDBInfo db, String tableName) {
       Map<String, TiTableInfo> tableMap = tableCache.get(db);
       if (tableMap == null) {
         tableMap = loadTables(db);
       }
-      return tableMap.get(tableName.toLowerCase());
+      TiTableInfo tbl = tableMap.get(tableName.toLowerCase());
+      if (tbl.isView()) return null;
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/catalog/Catalog.java]: should we add a TODO here?

**Pull request merged**